src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -367,7 +352,7 @@ public int Available
                         _nonBlockingConnectInProgress = false;
                     }
 
-                    if (_rightEndPoint == null)
+                    if (_rightEndPoint == null || !_isConnected)
```

Comment:
Without it, it will throw if RemoteEndPoint is accessed on copy and socket is not connected. It returns null on original socket in same scenario. 

```
  LocalEndPoint: 0.0.0.0:0
Unhandled exception. System.Net.Sockets.SocketException (57): Socket is not connected
   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, String callerName) in /Users/furt/github/wfurt-runtime/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs:line 5156
   at System.Net.Sockets.Socket.get_RemoteEndPoint() in /Users/furt/github/wfurt-runtime/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs:line 370
   at socket.Program.DumpInfo(Socket s) in /Users/furt/projects/socket/Program.cs:line 240
   at socket.Program.Main(String[] args) in /Users/furt/projects/socket/Program.cs:line 254
```

I tried to explain it in comment. This is because _rightEndPoint is _alwasys_ set on copy.